src/Internal/RejectedPromise.php

```diff
@@ -15,18 +16,36 @@
 final class RejectedPromise implements PromiseInterface
 {
     private $reason;
+    private $handled = false;
 
-    public function __construct(\Throwable $reason)
+    public function __construct(Throwable $reason)
     {
         $this->reason = $reason;
     }
 
+    public function __destruct()
+    {
+        if ($this->handled) {
+            return;
+        }
+
+        $message = 'Unhandled promise rejection with ';
+
+        $message .= get_class($this->reason) . ': ' . $this->reason->getMessage();
+        $message .= ' raised in ' . $this->reason->getFile() . ' on line ' . $this->reason->getLine();
+        $message .= PHP_EOL . $this->reason->getTraceAsString() . PHP_EOL;
+
+        fatalError($message);
```

Comment:
Rejections in `done()` are also unhandled rejections and should probably use the same error message format.

https://github.com/reactphp/promise/blob/57d86e562b74c940b5fc8487ec5116baa2799446/src/Internal/FulfilledPromise.php#L54

https://github.com/reactphp/promise/blob/master/src/Internal/RejectedPromise.php#L45

https://github.com/reactphp/promise/blob/master/src/Internal/RejectedPromise.php#L55

Maybe we can rename `fatalError()` to something like `unhandledRejectionError()` and generate the message there.